app/api/v3/position.py

```diff
@@ -448,6 +587,96 @@ def __init__(self):
         self.token_enabled = config.SHARE_TOKEN_ENABLED
         self.token_type = "IbetCoupon"
         self.token_model = CouponToken
+        self.idx_token_model = IDXCouponToken
+
+    def on_get_list_from_index(self, request_json: dict, session: Session, account_address: str):
+        offset = request_json["offset"]
+        limit = request_json["limit"]
+        include_token_details = True if request_json["include_token_details"] == "true" else False
+
+        # NOTE: Sub Query for sum of used amount
+        sub_tx_used = (
+            session.query(
+                sum_(IDXConsumeCoupon.amount).label("used"),
+                IDXConsumeCoupon.token_address,
+                IDXConsumeCoupon.account_address
+            ).
+            filter(IDXConsumeCoupon.account_address == account_address).
+            group_by(
+                IDXConsumeCoupon.token_address,
+                IDXConsumeCoupon.account_address
+            ).
+            subquery("sub_tx_used")
+        )
+        query = (
+            session.query(Listing.token_address, IDXPosition, self.idx_token_model, sub_tx_used.c.used).
+            join(
+                self.idx_token_model,
+                Listing.token_address == self.idx_token_model.token_address
+            ).
+            outerjoin(IDXPosition, and_(
+                Listing.token_address == IDXPosition.token_address,
+                IDXPosition.account_address == account_address
+            )).
+            outerjoin(sub_tx_used, and_(
+                Listing.token_address == sub_tx_used.c.token_address,
+                sub_tx_used.c.account_address == account_address
+            )).
+            filter(
+                or_(
+                    IDXPosition.balance != 0,
+                    IDXPosition.exchange_balance != 0,
+                    IDXPosition.exchange_commitment != 0,
+                    sub_tx_used.c.used != 0,
+                    session.query(IDXTransfer).filter(
+                        and_(
+                            IDXTransfer.token_address == Listing.token_address,
+                            IDXTransfer.to_address == account_address
+                        )
+                    ).exists()
+                )
+            ).order_by(Listing.id)
+        )
```

Comment:
Unlike other token type, coupon API handles IDXTransfer and returns "used" amount in current.
So sub query is required here.